src/Common/src/CoreLib/System/Buffers/ConfigurableArrayPool.cs

```diff
@@ -250,9 +252,12 @@ internal void Return(T[] array)
                 {
                     _lock.Enter(ref lockTaken);
 
-                    if (_index != 0)
+                    int index = _index;
+                    if (index != 0)
                     {
-                        _buffers[--_index] = array;
+                        index--;
+                        _buffers[index] = array;
+                        _index--;
```

Comment:
This is a bug. The index should not be decremented twice.

Reply:
I tried that as well but it turns out the codegen is worse in this case
```asm
// _index = index;
L0081: mov edx, ebx
L0083: call 0x71d36700
L0088: mov ecx, [ebp-0x2c]
L008b: mov [ecx+0xc], ebx

// _index--
L0081: call 0x71d36700
L0086: mov ecx, [ebp-0x2c]
L0089: dec dword [ecx+0xc]
```


I will open the PR in coreclr